File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        max_sub = ""
        for i in range(len(num) - 2):
            if num[i] == num[i+1] == num[i+2]:
                candidate = num[i:i+3]
                if candidate > max_sub:
                    max_sub = candidate
        return max_sub

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string max_sub = "";
        for (int i=0; i <= (int)num.size() - 3; i++) {
            if (num[i] == num[i+1] && num[i] == num[i+2]) {
                string candidate = num.substr(i,3);
                if (candidate > max_sub) {
                    max_sub = candidate;
                }
            }
        }
        return max_sub;
    }
};